pkg/prompb/remote.pb.go

```diff
@@ -946,6 +975,40 @@ func (m *WriteRequest) Unmarshal(dAtA []byte) error {
 				return err
 			}
 			iNdEx = postIndex
+		case 3:
+			if wireType != 2 {
+				return fmt.Errorf("proto: wrong wireType = %d for field Metadata", wireType)
+			}
+			var msglen int
+			for shift := uint(0); ; shift += 7 {
+				if shift >= 64 {
+					return ErrIntOverflowRemote
+				}
+				if iNdEx >= l {
+					return io.ErrUnexpectedEOF
+				}
+				b := dAtA[iNdEx]
+				iNdEx++
+				msglen |= int(b&0x7F) << shift
+				if b < 0x80 {
+					break
+				}
+			}
+			if msglen < 0 {
+				return ErrInvalidLengthRemote
+			}
+			postIndex := iNdEx + msglen
+			if postIndex < 0 {
+				return ErrInvalidLengthRemote
+			}
+			if postIndex > l {
+				return io.ErrUnexpectedEOF
+			}
+			m.Metadata = append(m.Metadata, MetricMetadata{})
```

Comment:
We should implement the same memory optimization as in https://github.com/timescale/promscale/commit/a565b9270db7ccb6a65a7c262e689834a57a17d1
```suggestion
			if len(m.Metadata) < cap(m.Metadata) {
				m.Metadata = m.Metadata[:len(m.Metadata)+1]
				m.Metadata[len(m.Metadata)-1].Reset()
			} else {
				m.Metadata = append(m.Metadata, MetricMetadata{})
			}
```